plugins/events/index.js

```diff
@@ -40,16 +40,18 @@ const eventsPlugin = {
                 });
             }
 
-            // Add user as admin if permissions good and does not already exist
-            if (!pipeline.admins[username]) {
-                newAdmins[username] = true;
-                // This is needed to make admins dirty and update db
-                pipeline.admins = newAdmins;
+            // Put current user at the head of admins to use its SCM token after this
+            // SCM token is got from the first pipeline admin
+            const newAdminNames = [username, ...Object.keys(pipeline.admins)];
+            const newAdmins = {};
 
-                return pipeline.update();
-            }
+            newAdminNames.forEach(name => {
+                newAdmins[name] = true;
```

Comment:
`username` is string, so `Object.keys` order is insertion order.
https://262.ecma-international.org/6.0/#sec-ordinary-object-internal-methods-and-internal-slots-ownpropertykeys